### PR TITLE
bugfix: fix ta grades view

### PR DESCRIPTION
**Pull request by pretendWhale**
## Motivation and Context

Currently the TA view for grades on grade entry forms does load due to an incorrect query. This change fixes that query and re-enables the TA view for grades on GEFs 

## Your Changes

**Description**:
Corrects query on grade entry forms `populate_grades_table` method for TAs and adds tests for `populate_grades_table` method

**Type of change** (select all that apply):

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

Tested via the web view to ensure the grade table loads properly

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes.

### CODE REVIEW

File: spec/controllers/grade_entry_forms_controller_spec.rb [hunk 1]
```diff
@@ -566,4 +566,41 @@
       include_examples 'switch assignment tests'
     end
   end
+
+  describe '#populate_grade_table' do
```
Comment (mishaschwartz) [inline spec/controllers/grade_entry_forms_controller_spec.rb:570]: `populate_grades_table` (s is missing after grade)

File: spec/controllers/grade_entry_forms_controller_spec.rb [hunk 2]
```diff
@@ -566,4 +566,41 @@
...
+    before(:each) do
+      get_as role, :populate_grades_table, params: { course_id: course.id, id: grade_entry_form_with_data.id }
+    end
+    context 'an instructor' do
```
Comment (mishaschwartz) [inline spec/controllers/grade_entry_forms_controller_spec.rb:574]: add a test that makes sure they can see _all_ the student data (not just some number > 0)

File: spec/controllers/grade_entry_forms_controller_spec.rb [hunk 3]
```diff
@@ -566,4 +566,41 @@
...
+        it 'returns data' do
+          expect(response.parsed_body['data'].length).to eq 0
+        end
+      end
+      context 'who has been assigned a student' do
+        let(:student) do
+          grade_entry_form_with_data.grade_entry_students.joins(role: :end_user).find_by('users.user_name': 'c8shosta')
+        end
+        let!(:grade_entry_student_ta) { create(:grade_entry_student_ta, ta: role, grade_entry_student: student) }
+        context 'who has not been assigned a student' do
```
Comment (mishaschwartz) [inline spec/controllers/grade_entry_forms_controller_spec.rb:597]: Did you mean to put this context here? The title of it is confusing

File: spec/controllers/grade_entry_forms_controller_spec.rb [hunk 4]
```diff
@@ -566,4 +566,41 @@
...
+          it 'returns data' do
+            get_as role, :populate_grades_table, params: { course_id: course.id, id: grade_entry_form_with_data.id }
+            expect(response.parsed_body['data'].length).to be > 0
```
Comment (mishaschwartz) [inline spec/controllers/grade_entry_forms_controller_spec.rb:600]: Add at least another test to make sure they can see the right students data and the right number of students (not just some amount of data > 0)

File: spec/controllers/grade_entry_forms_controller_spec.rb (hunk 1 above)
Comment (mishaschwartz) [inline spec/controllers/grade_entry_forms_controller_spec.rb:570]: Please add a few more tests to make sure that the correct data is being sent, not just that *some* data is being sent.

**Review by mishaschwartz — CHANGES_REQUESTED** [review]

File: spec/controllers/grade_entry_forms_controller_spec.rb (hunk 1 above)
Comment (mishaschwartz) [reply spec/controllers/grade_entry_forms_controller_spec.rb]: keys are fine :)

File: spec/controllers/grade_entry_forms_controller_spec.rb [hunk 5]
```diff
@@ -566,4 +566,58 @@
...
+    context 'an instructor' do
+      let(:data_fields) { %w[_id released_to_student user_name first_name last_name hidden section_id] }
+      it 'returns a 200 response' do
+        expect(response.status).to eq 200
+      end
+      it 'returns data' do
+        expect(response.parsed_body['data'].length).to be > 0
+      end
+      it 'returns all student data' do
+        expect(response.parsed_body['data'].length).to eq Student.count
```
Comment (mishaschwartz) [inline spec/controllers/grade_entry_forms_controller_spec.rb:583]: I'm trying to make sure that tests are more future-proof for when we add tests with multiple courses so can this be `course.students.count`

**Review by mishaschwartz — CHANGES_REQUESTED** [review]

File: spec/controllers/grade_entry_forms_controller_spec.rb (hunk 5 above)
Comment (pretendWhale) [reply spec/controllers/grade_entry_forms_controller_spec.rb]: right, duh

**Review by mishaschwartz — APPROVED** [review]

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**